treeherder/perf/backfill_tool.py

```diff
@@ -0,0 +1,46 @@
+import logging
+
+from django.core.exceptions import ObjectDoesNotExist
+
+from treeherder.model.models import Job
+from treeherder.perf.exceptions import CannotBackfill
+from treeherder.services.taskcluster import TaskclusterModel
+
+logger = logging.getLogger(__name__)
+
+
+class BackfillTool:
+
+    def __init__(self, taskcluster_model: TaskclusterModel):
+        self.tc_model = taskcluster_model
+
+    def backfill_job(self, job_id: str) -> str:
+        job = self._fetch_job(job_id)
+
+        self.assert_backfill_ability(job)
+
+        logger.debug(f"Fetching decision task of job {job.id}...")
+        task_id_to_backfill = job.taskcluster_metadata.task_id
+        decision_job = job.fetch_associated_decision_job()
+        decision_task_id = decision_job.taskcluster_metadata.task_id
+
+        logger.debug(f"Requesting backfill for task {task_id_to_backfill}...")
+        task_id = self.tc_model.trigger_action(
+            action='backfill',
+            task_id=task_id_to_backfill,
+            decision_task_id=decision_task_id,
+            input={},
+            root_url=job.repository.tc_root_url
+        )
+        return task_id
+
+    def assert_backfill_ability(self, over_job: Job):
+        if over_job.repository.is_try_repo:
+            raise CannotBackfill("Try repository isn't suited for backfilling.")
```

Comment:
shouldn't this be autoland only?  what about other restrictions like 'linux' or certain jobs/suites?

Reply:
> shouldn't this be autoland only?

We're doing this kind of check at the [edge of the management script](https://github.com/mozilla/treeherder/blob/5e883dd9e813325acf60ee8602769ce000b132b3/treeherder/perf/management/commands/synthesize_backfill_report.py#L67-L69).
We're actually including `mozilla-beta` as well, as that's also part of the sheriffing process.

> what about other restrictions like 'linux' or certain jobs/suites?

The client *([PerfSheriffBot](https://github.com/mozilla/treeherder/blob/313f73df14d6b922b21f1212c1b7c403a76f18e3/treeherder/perf/perf_sheriff_bot.py#L13))* will be responsible for this kind of check.

Reply:
why is this check for try in here and not at the edge?  beta is ok, but when is there a need to backfill?  we run all the tests on every commit for mozilla-beta.

Reply:
> why is this check for try in here and not at the edge?

Because it is reasonable to assume, even at this level, that backfilling cannot be done for try pushes.

Reply:
> beta is ok, but when is there a need to backfill? we run all the tests on every commit for mozilla-beta.

Fair point. Still, on next iterations of the backfill bot, we want our backfills to include retriggers  as well. These are still relevant on `mozilla-beta`, to help narrow down the suspect patches.

Reply:
are there other checks planned?  this seems overkill to have a full method to ensure that try is not the repo?